slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py

```diff
@@ -56,268 +66,240 @@ def icon(self, name='SegmentEditorEffect.png'):
         logging.debug('Update Segmentation Mask from: {}'.format(in_file))
         if in_file is None or os.path.exists(in_file) is False:
             return False
 
         segmentationNode = self.scriptedEffect.parameterSetNode().GetSegmentationNode()
-        selectedSegmentId = self.scriptedEffect.parameterSetNode().GetSelectedSegmentID()
-
         segmentation = segmentationNode.GetSegmentation()
-        segment = segmentation.GetSegment(selectedSegmentId)
-        color = segment.GetColor()
-        label = segment.GetName()
+        currentSegment = self.currentSegment()
 
         labelImage = sitk.ReadImage(in_file)
         labelmapVolumeNode = sitkUtils.PushVolumeToSlicer(labelImage, None, className='vtkMRMLLabelMapVolumeNode')
-        labelmapVolumeNode.SetName(label)
-        # [success, labelmapVolumeNode] = slicer.util.loadLabelVolume(in_file, {'name': label}, returnNode=True)
-
-        logging.debug('Removing temp segmentation with id: {} with color: {}'.format(selectedSegmentId, color))
-        segmentationNode.RemoveSegment(selectedSegmentId)
-
-        originalSegments = dict()
-        for i in range(segmentation.GetNumberOfSegments()):
-            segmentId = segmentation.GetNthSegmentID(i)
-            originalSegments[segmentId] = i
 
+        numberOfExistingSegments = segmentation.GetNumberOfSegments()
         slicer.modules.segmentations.logic().ImportLabelmapToSegmentationNode(labelmapVolumeNode, segmentationNode)
         slicer.mrmlScene.RemoveNode(labelmapVolumeNode)
 
-        addedSegments = 0
-        modelLabels = modelInfo.get('labels')
-        for i in range(segmentation.GetNumberOfSegments()):
-            segmentId = segmentation.GetNthSegmentID(i)
+        modelLabels = modelInfo['labels']
+        numberOfAddedSegments = segmentation.GetNumberOfSegments() - numberOfExistingSegments
+        logging.debug('Adding {} segments'.format(numberOfAddedSegments))
+        addedSegmentIds = [segmentation.GetNthSegmentID(numberOfExistingSegments + i) for i in range(numberOfAddedSegments)]
+        for i, segmentId in enumerate(addedSegmentIds):
             segment = segmentation.GetSegment(segmentId)
-            if originalSegments.get(segmentId) is not None:
-                logging.debug('No change for existing segment with id: {} => {}'.format(segmentId, segment.GetName()))
-                continue
-
-            logging.debug('Setting new segmentation with id: {} => {}'.format(segmentId, segment.GetName()))
-            if addedSegments == 0:
-                segment.SetColor(color)
-                segment.SetName(label)
-
-                self.scriptedEffect.parameterSetNode().SetSelectedSegmentID(segmentId)
-                logging.debug('Extreme Points: {}'.format(extreme_points))
-                if extreme_points is not None:
-                    segment.SetTag("DExtr3DExtremePoints", json.dumps(extreme_points))
-
+            if i==0 and overwriteCurrentSegment and currentSegment:
```

Comment:
i guess some issue with this condition... not modifying the current selected segment..

Reply:
The current behavior is intentional. See https://github.com/NVIDIA/ai-assisted-annotation-client/pull/47#issuecomment-566203912